File: examples/de.genesez.example.java4.forms/src-gen/forms/Rectangle.java
package forms;

import java.util.*;

/* PROTECTED REGION ID(java.type.import._16_0_b6f02e1_1249464012328_770017_760) ENABLED START */
// TODO: put your further include + require statements here
/* PROTECTED REGION END */

/**
 * Describes the geometrical form <b>rectangle</b> as a specic <i>polygon</i>
 * @author	dreamer
 */
public class Rectangle extends Polygon {
	
	// -- generated constructors --------------------------------------------
	/**
	 * constructs a rectangle by the 2 diagonally points
	 * @generated	constructor for class '<em><b>Rectangle</b></em>'
	 * @param	a	the lower left point of the rectangle
	 * @param	c	the upper right point of the rectangle
	 */
	public Rectangle(Point a, Point c) {
		/* PROTECTED REGION ID(java.constructor._16_0_b6f02e1_1249464153812_196550_828) ENABLED START */
		super(new Point[] {
		a, new Point(c.getX(), a.getY()), c, new Point(a.getX(), c.getY())
		});
		/* PROTECTED REGION END */
	}
	
	// -- generated method stubs for implementations + derived attributes ---
	
	// -- own code implementation -------------------------------------------
	/* PROTECTED REGION ID(java.class.own.code.implementation._16_0_b6f02e1_1249464012328_770017_760) ENABLED START */
	// TODO: put your own implementation code here
	/* PROTECTED REGION END */
}

File: examples/branches/GeneSEZ_Examples_2-1-0/org.genesez.example.php.forms/src-gen/Form/Rectangle.php
<?php
require_once 'Form/Point.php';
require_once 'Form/Polygon.php';

/* PROTECTED REGION ID(php.own.imports._16_0_b6f02e1_1249464012328_770017_760) ENABLED START */
// TODO: put your further include + require statements here
/* PROTECTED REGION END */

/**
 * Describes the geometrical form <b>rectangle</b> as a specic <i>polygon</i>
 * @see		Form_Polygon
 * @author	nicher
 * @package	de.genesez.example.java.forms
 */
class Form_Rectangle extends Form_Polygon  {
	
	
	// -- constructors + destructors ----------------------------------------
	
	/**
	 * constructs a rectangle by the 2 diagonally points
	 * @param	Form_Point	$a	the lower left point of the rectangle
	 * @param	Form_Point	$c	the upper right point of the rectangle
	 */
	public function __construct(Form_Point $a, Form_Point $c) {
		/* PROTECTED REGION ID(php.constructor._16_0_b6f02e1_1249464153812_196550_828) ENABLED START */
		parent::__construct(array($a, new Form_Point($c->x, $a->y), $c, new Form_Point($a->x, $c->y)));
		/* PROTECTED REGION END */
	}
	
	
	// -- method implementations --------------------------------------------
	

	
	
	
	// -- own code implementation -------------------------------------------
	/* PROTECTED REGION ID(php.class.own.code.implementation._16_0_b6f02e1_1249464012328_770017_760) ENABLED START */
	// TODO: put your further code implementations for class 'forms_Rectangle' here
	/* PROTECTED REGION END */
}
?>
